instrumentation/github.com/gorilla/mux/otelmux/test/mux_test.go

```diff
@@ -69,6 +73,33 @@ func TestSDKIntegration(t *testing.T) {
 	)
 }
 
+func TestNotFoundIsNotError(t *testing.T) {
+	sr := tracetest.NewSpanRecorder()
+	provider := sdktrace.NewTracerProvider()
+	provider.RegisterSpanProcessor(sr)
+
+	router := mux.NewRouter()
+	router.Use(otelmux.Middleware("foobar", otelmux.WithTracerProvider(provider)))
+	router.HandleFunc("/does/not/exist", notfound)
+
+	r0 := httptest.NewRequest("GET", "/does/not/exist", nil)
+	w := httptest.NewRecorder()
+	router.ServeHTTP(w, r0)
+
+	require.Len(t, sr.Ended(), 1)
+	assertSpan(t, sr.Ended()[0],
+		"/does/not/exist",
+		trace.SpanKindServer,
+		attribute.String("http.server_name", "foobar"),
+		attribute.Int("http.status_code", http.StatusNotFound),
+		attribute.String("http.method", "GET"),
+		attribute.String("http.target", "/does/not/exist"),
+		attribute.String("http.route", "/does/not/exist"),
+	)
+	assert.NotEqual(t, sr.Ended()[0].Status().Code, codes.Error)
```

Comment:
Should this assert that the code is equal to `codes.Unset`?  Pretty sure that instrumentation like this is never supposed to set `codes.Ok` and that's the only other option.

Reply:
I agree that it doesn't make any sense to return `codes.Ok` but it's not explicitly stated in the `semconv` docs that it won't. I went for this because it DOES explicitly state:
```
Exclude 4xx for SERVER to set the appropriate status.
```
I'm happy to update it if you think it's preferable to `assert.Equal(..., codes.Unset)`.

Reply:
Updated the test, and caught up with main